packages/kbn-config/src/config_service.ts

```diff
@@ -177,6 +177,25 @@ export class ConfigService {
     // if plugin hasn't got a config schema, we try to read "enabled" directly
     const isEnabled = validatedConfig?.enabled ?? config.get(enabledPath);
 
+    // if we implicitly added an `enabled` config to a plugin without a schema,
+    // we log a deprecation warning, as this will not be supported in 8.0
+    if (validatedConfig?.enabled === undefined && isEnabled !== undefined) {
+      const deprecationPath = pathToString(enabledPath);
+      const deprecatedConfigDetails: DeprecatedConfigDetails = {
+        title: `Setting "${deprecationPath}" is deprecated`,
+        message: `Configuring "${deprecationPath}" is deprecated and will be removed in 8.0.0.`,
+        correctiveActions: {
+          manualSteps: [
+            `Remove "${deprecationPath}" from the Kibana config file, CLI flag, or environment variable (in Docker only) before upgrading to 8.0.0.`,
```

Comment:
Do we have any guidance on the version naming? Why not `...before upgrading to v8.0.0 release`?

Reply:
I had considered enforcing semver here and actually validating at runtime to ensure it was the expected format, but wasn't sure if it was worth it for something that ultimately is just printing to the logs. WDYT?